zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -23,61 +23,28 @@ import zio.http.netty.NettyConfig
 import zio.http.netty.client._
 
 import java.net.{InetSocketAddress, URI} // scalafix:ok;
-
-trait ZClient[-Env, -In, +Err, +Out] extends HeaderOps[ZClient[Env, In, Err, Out]] { self =>
-  def headers: Headers
-
-  def method: Method
-
-  def sslConfig: Option[ClientSSLConfig]
-
-  def url: URL
-
-  def version: Version
+import java.net.MalformedURLException
+
+final case class ZClient[-Env, -In, +Err, +Out](
+  version: Version,
+  url: URL,
+  headers: Headers,
+  sslConfig: Option[ClientSSLConfig],
+  bodyEncoder: ZClient.BodyEncoder[Env, Err, In],
+  bodyDecoder: ZClient.BodyDecoder[Env, Err, Out],
+  driver: ZClient.Driver[Env, Err],
+) extends HeaderOps[ZClient[Env, In, Err, Out]] { self =>
+  def apply(request: Request)(implicit ev: Body <:< In, trace: Trace): ZIO[Env & Scope, Err, Out] =
+    self.request(request)
```

Comment:
In term of API design, I'm personally not a fan of the fact that the apply method triggers the HTTP requests.
It leads to code looking like this:
```
val client: Client = ...
val request: Request = ...
val response = client(request)
```

It's kind of odd, I think. It's IMO not really idiomatic Scala.
I'm also not a big fan of the `client.request` method name now that the request concept has been reified into the `Request` type. It leads to code like this `client.request(request)`. Maybe `client.call(request)` could be better